### ISSUE TITLE
Usecase: Rewind/Fill/Prime via the pump (reset status indicators)

### ISSUE THREAD

**Issue by rustymonkey** [MilosKozak/AndroidAPS#2138]
In AAPS 2.4 RC4 it's possible to initiate a Pump Site Change and/or an Insulin Cartridge change which resets the Canula Age and the Reservoir Age respectively. In #2080, the usecase of performing these operations solely via the pump was addresed in terms of resetting autosense, however, it would be ideal if AAPS could also:
- Reset the Canula age each time a Fixed Prime operation is performed on the pump [already implemented although there are false resets recorded--see case 2) below]
- Reset the Reservoir age each time a Rewind/Fill operation is performed on the pump [not implemented--see cases 2/3 below]

The current behavior (with a Medtronic 722) is that:
1) When performing Fixed Prime only (e.g. for a cannula change without changing the reservoir): CAN age is automatically reset/Treatments indicate Pump site change (as expected).
2) When performing Rewind/Fill only (e.g. to fill the reservoir but re-use the existing canula): INS age isn't reset. CAN age is reset/Treatments indicated Pump Site Change even though the canula wasn't changed.
3) When performing Fixed Prime and Rewind/Fill: CAN age is automatically reset, but INS age is not.

**Comment by Webifi** [comment]
I'd rather NOT have the cannula age reset when changing/refilling a reservoir.   Currently, that seems to be happening on 2.6 RC6.

Often, I'll use up my reservoir before my cannula has reached, or is just at EOL, and I'll give it a few more units -- especially if I'm on the road and not in a good place to replace the cannula.   When I do this, I don't want to reset the age of the cannula, else I could mistakenly leave it in a bit too long -- as I just found myself doing this past week. 

Perhaps it's possible to detect if a prime was initiated on the pump, and only reset the cannula in that case, but not just for a rewind/fill?

**Comment by rustymonkey** [comment]
I agree. I mentioned this in case 2. outlined above: that when a rewind/fill is performed INS age isn't reset but CAN age is, when in fact, it should be the opposite. But I probably could have made it more clear.

**Comment by andyrozman** [comment]
I think this was fixed now on newest dev. Can you please check if it works correctly or if we require additional changes...

**Comment by Webifi** [comment]
@andyrozman By "newest dev", do you mean the dev branch that hasn't been updated in three months?

**Comment by andyrozman** [comment]
This repository was abandoned about 3 months ago (see README)... There is new repository under nightscout/AndroidAPS, where dev is updated daily...

**Comment by Webifi** [comment]
Thanks.  However, I could be blind, but I don't see a note about that in any of the readme files on any of the MilosKozak branches.

**Comment by p5nbTgip0r** [comment]
> I think this was fixed now on newest dev. Can you please check if it works correctly or if we require additional changes...

I didn't even think to check the issues for this repository, sorry for not referencing this issue. This issue describes the exact problems I focused on with my pull request. I tested it with my most recent reservoir+cannula change and this issue appears to be fixed but I would appreciate more testing.

**Comment by andyrozman** [comment]
My solution if this doesn't work in all cases (only scenario where this wouldn't work, would be when Init is done from AAPS and not on pump itself), would be to give user option to ignore both this items (Rewind and Prime) from history, if they set it so in configuration... @rustymonkey 

@Webifi Sorry, I thought this information was put into README

**Comment by Webifi** [comment]
I can confirm that refilling a reservoir was correctly recorded, and did not improperly record a cannula change.
